test/rules/array-type/array-simple/test.ts.lint

```diff
@@ -0,0 +1,50 @@
+let x: Array<number> = [1] as number[];
+       ~~~~~~~~~~~~~                    [Array type using 'Array<T>' is forbidden for simple types. Use 'T[]' instead.]
+let y: string[] = <Array<string>>["2"];
+                   ~~~~~~~~~~~~~        [Array type using 'Array<T>' is forbidden for simple types. Use 'T[]' instead.]
+let z: Array = [3, "4"];
+       ~~~~~             [Array type using 'Array<T>' is forbidden for simple types. Use 'T[]' instead.]
+
+let xx: Array<Array<number>> = [[1, 2], [3]];
+        ~~~~~~~~~~~~~~~~~~~~                  [Array type using 'Array<T>' is forbidden for simple types. Use 'T[]' instead.]
+              ~~~~~~~~~~~~~                   [Array type using 'Array<T>' is forbidden for simple types. Use 'T[]' instead.]
+let yy: number[][] = [[4, 5], [6]];
+
+type Arr<T> = Array<T>;
+              ~~~~~~~~  [Array type using 'Array<T>' is forbidden for simple types. Use 'T[]' instead.]
+
+// Ignore user defined aliases
+let yyyy: Arr<Array<Arr<string>>[]> = [[[["2"]]]];
+              ~~~~~~~~~~~~~~~~~~                   [Array type using 'Array<T>' is forbidden for simple types. Use 'T[]' instead.]
+
+interface ArrayClass<T> {
+    foo: Array<T>;
+         ~~~~~~~~  [Array type using 'Array<T>' is forbidden for simple types. Use 'T[]' instead.]
```

Comment:
might be nice to also have a dotted name, like
`namespace a {
  class B {}
}`
`Array<a.B>` should be disallowed